keras_core/layers/layer.py

```diff
@@ -293,12 +293,19 @@ def _initializer_tracker(self):
                 ),
             }
         )
+        # Remove attribute tracking for lists (TF-specific attribute)
```

Comment:
This class shouldn't have anything that's TF-specific.

Could we leverage the `no_automatic_dependency_tracking` from `tracking.py` here?

Or otherwise handle this in the TF-specific Layer class in the backend?

Reply:
The primary issue is that SavedModel is completely independent of our tracking suite and follows TF's tracking suite. @tracking.no_automatic_dependency_tracking is already on this function but is not removing dependency tracking for SavedModel. We also can't handle this in the TF-specific Layer class because this directly pertains to the lists of trainable_variables, non_trainable_variables, seed_generators shown a few lines above and the issues they cause with TF tracking. Therefore, the only options (without refactoring a lot of the code) are either using this attribute as shown or applying a TF-specific decorator (which adds a TF dependency here).

This attribute does nothing in the other backends and is only TF-relevant. We can even delete it after its usage here.

Reply:
Let's keep it here, but use a `if backend.backend() == "tensorflow"` block so we can easily identify backend specific special cases in the codebase. Also enables you to get rid of the getattr.

Reply:
Got it, added the block for special casing. Fixed